Investigate and fix failing CI on Toil master branch

This seems to only be failing for integration tests: https://ucsc-ci.com/databiosphere/toil/-/pipelines/7423

┆Issue is synchronized with this [Jira Story](https://ucsc-cgl.atlassian.net/browse/TOIL-1592)
┆Issue Number: TOIL-1592

```
=================================== FAILURES ===================================
[31m[1m___________________ AWSProvisionerBenchTest.test_AMI_finding ___________________[0m
[gw2] linux -- Python 3.12.0 /builds/databiosphere/toil/venv/bin/python
Traceback (most recent call last):
  File "/usr/lib/python3.12/unittest/case.py", line 58, in testPartExecutor
    yield
  File "/usr/lib/python3.12/unittest/case.py", line 634, in run
    self._callTestMethod(testMethod)
  File "/usr/lib/python3.12/unittest/case.py", line 589, in _callTestMethod
    if method() is not None:
       ^^^^^^^^
  File "/builds/databiosphere/toil/src/toil/test/provisioners/aws/awsProvisionerTest.py", line 54, in test_AMI_finding
    provisioner = AWSProvisioner('fakename', 'mesos', zone, 10000, None, None)
                  ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
TypeError: AWSProvisioner.__init__() missing 1 required positional argument: 'enable_fuse'
```
https://ucsc-ci.com/databiosphere/toil/-/jobs/71590/raw

```

=================================== FAILURES ===================================
[31m[1m____________ ToilWESServerWorkflowTest.test_run_workflow_https_url _____________[0m
[gw0] linux -- Python 3.12.0 /builds/databiosphere/toil/venv/bin/python
Traceback (most recent call last):
  File "/usr/lib/python3.12/unittest/case.py", line 58, in testPartExecutor
    yield
  File "/usr/lib/python3.12/unittest/case.py", line 634, in run
    self._callTestMethod(testMethod)
  File "/usr/lib/python3.12/unittest/case.py", line 589, in _callTestMethod
    if method() is not None:
       ^^^^^^^^
  File "/builds/databiosphere/toil/src/toil/test/server/serverTest.py", line 590, in test_run_workflow_https_url
    self._wait_for_success(client, run_id)
  File "/builds/databiosphere/toil/src/toil/test/server/serverTest.py", line 463, in _wait_for_success
    self._wait_for_status(client, run_id, "COMPLETE")
  File "/builds/databiosphere/toil/src/toil/test/server/serverTest.py", line 454, in _wait_for_status
    self._report_log(client, run_id)
  File "/builds/databiosphere/toil/src/toil/test/server/serverTest.py", line 380, in _report_log
    self._report_absolute_url(client, stdout)
  File "/builds/databiosphere/toil/src/toil/test/server/serverTest.py", line 391, in _report_absolute_url
    raise RuntimeError("Could not find place to break URL: " + url)
RuntimeError: Could not find place to break URL: 
```
https://ucsc-ci.com/databiosphere/toil/-/jobs/71610/raw

--- a/src/toil/lib/aws/__init__.py
+++ b/src/toil/lib/aws/__init__.py
@@ -137,12 +137,13 @@ def get_current_aws_zone() -> Optional[str]:
     Finally, if we have boto2, and a default region is configured in Boto 2,
     chooses a zone in that region.
 
-    Returns None if no method can produce a zone to use.
+    Returns 'us-east-1a' if no method can produce a zone to use.
     """
     return get_aws_zone_from_environment() or \
         get_aws_zone_from_metadata() or \
         get_aws_zone_from_environment_region() or \
-        get_aws_zone_from_boto()
+        get_aws_zone_from_boto() or \
+        'us-east-1a'  # AWS's native default
 
 def zone_to_region(zone: str) -> AWSRegionName:
     """Get a region (e.g. us-west-2) from a zone (e.g. us-west-1c)."""

--- a/src/toil/lib/aws/utils.py
+++ b/src/toil/lib/aws/utils.py
@@ -203,7 +203,7 @@ def create_s3_bucket(
 
     *ALL* S3 bucket creation should use this function.
     """
-    logger.debug("Creating bucket '%s' in region %s.", bucket_name, region)
+    logger.info("Creating bucket '%s' in region %s.", bucket_name, region)
     if region == "us-east-1":  # see https://github.com/boto/boto3/issues/125
         bucket = s3_resource.create_bucket(Bucket=bucket_name)
     else:

--- a/src/toil/server/wes/tasks.py
+++ b/src/toil/server/wes/tasks.py
@@ -112,7 +112,7 @@ class ToilWorkflowRunner:
             logger.info("Linking workflow from filesystem.")
             link_file(src=src_url[7:], dest=dest)
         elif src_url.startswith(("http://", "https://")):
-            logger.info("Downloading workflow_url from the Internet.")
+            logger.info(f"Downloading workflow_url from the Internet.  src: {src_url} dst: {dest}")
             download_file_from_internet(src=src_url, dest=dest, content_type="text/")
         elif src_url.startswith("s3://"):
             logger.info("Downloading workflow_url from Amazon S3.")

--- a/src/toil/test/provisioners/aws/awsProvisionerTest.py
+++ b/src/toil/test/provisioners/aws/awsProvisionerTest.py
@@ -51,7 +51,7 @@ class AWSProvisionerBenchTest(ToilTest):
     @needs_aws_ec2
     def test_AMI_finding(self):
         for zone in ['us-west-2a', 'eu-central-1a', 'sa-east-1b']:
-            provisioner = AWSProvisioner('fakename', 'mesos', zone, 10000, None, None)
+            provisioner = AWSProvisioner('fakename', 'mesos', zone, 10000, None, None, enable_fuse=False)
             ami = provisioner._discoverAMI()
             # Make sure we got an AMI and it looks plausible
             assert(ami.startswith('ami-'))
@@ -62,7 +62,7 @@ class AWSProvisionerBenchTest(ToilTest):
         Make sure the `_write_file_to_cloud()` and `_read_file_from_cloud()`
         functions of the AWS provisioner work as intended.
         """
-        provisioner = AWSProvisioner(f'aws-provisioner-test-{uuid4()}', 'mesos', 'us-west-2a', 50, None, None)
+        provisioner = AWSProvisioner(f'aws-provisioner-test-{uuid4()}', 'mesos', 'us-west-2a', 50, None, None, enable_fuse=False)
         key = 'config/test.txt'
         contents = b"Hello, this is a test."
 

--- a/src/toil/test/server/serverTest.py
+++ b/src/toil/test/server/serverTest.py
@@ -223,9 +223,7 @@ class BucketUsingTest(ToilTest):
         super().tearDownClass()
 
 class AWSStateStoreTest(hidden.AbstractStateStoreTest, BucketUsingTest):
-    """
-    Test AWS-based state storage.
-    """
+    """Test AWS-based state storage."""
 
     from toil.server.utils import AbstractStateStore
 
@@ -345,7 +343,6 @@ class AbstractToilWESServerTest(ToilTest):
         """
         rv = self._fetch_run_log(client, run_id)
         logger.debug('Log info: %s', rv.json)
-        self.assertIn("run_log", rv.json)
         run_log = rv.json.get("run_log")
         self.assertEqual(type(run_log), dict)
         if "exit_code" in run_log:
@@ -363,22 +360,15 @@ class AbstractToilWESServerTest(ToilTest):
             self.assertEqual(task_log.get("exit_code"), 0)
 
     def _report_log(self, client: "FlaskClient", run_id: str) -> None:
-        """
-        Report the log for the given workflow run.
-        """
+        """Report the log for the given workflow run."""
         rv = self._fetch_run_log(client, run_id)
-        self.assertIn("run_log", rv.json)
+        logger.critical(f'{rv.json}')
         run_log = rv.json.get("run_log")
         self.assertEqual(type(run_log), dict)
-        self.assertIn("stdout", run_log)
-        stdout = run_log.get("stdout")
-        self.assertEqual(type(stdout), str)
-        self.assertIn("stderr", run_log)
-        stderr = run_log.get("stderr")
-        self.assertEqual(type(stderr), str)
-        logger.info("Got stdout %s and stderr %s", stdout, stderr)
-        self._report_absolute_url(client, stdout)
-        self._report_absolute_url(client, stderr)
+        self.assertEqual(type(run_log.get("stdout")), str)
+        self.assertEqual(type(run_log.get("stderr")), str)
+        self._report_absolute_url(client, run_log.get("stdout"))
+        self._report_absolute_url(client, run_log.get("stderr"))
 
     def _report_absolute_url(self, client: "FlaskClient", url: str):
         """
@@ -574,8 +564,7 @@ class ToilWESServerWorkflowTest(AbstractToilWESServerTest):
         """Test run example CWL workflow from the Internet."""
         with self.app.test_client() as client:
             rv = client.post("/ga4gh/wes/v1/runs", data={
-                "workflow_url": "https://raw.githubusercontent.com/DataBiosphere/toil/releases/5.4.x/src/toil"
-                                "/test/cwl/echo.cwl",
+                "workflow_url": "https://raw.githubusercontent.com/DataBiosphere/toil/4cb5bb3871ac21a9793f638b83775926ed94a226/src/toil/test/cwl/echo.cwl",
                 "workflow_type": "CWL",
                 "workflow_type_version": "v1.2",
                 "workflow_params": json.dumps({"message": "Hello, world!"}),

--- a/src/toil/test/server/serverTest.py
+++ b/src/toil/test/server/serverTest.py
@@ -362,7 +362,7 @@ class AbstractToilWESServerTest(ToilTest):
     def _report_log(self, client: "FlaskClient", run_id: str) -> None:
         """Report the log for the given workflow run."""
         rv = self._fetch_run_log(client, run_id)
-        logger.critical(f'{rv.json}')
+        logger.debug(f'Report log response: {rv.json}')
         run_log = rv.json.get("run_log")
         self.assertEqual(type(run_log), dict)
         self.assertEqual(type(run_log.get("stdout")), str)

